RELEASE.md

```diff
@@ -1,18 +1,69 @@
 # Falco Release Process
 
-Our release process is mostly automated, but we still need some manual steps to initiate and complete it.
 
-Changes and new features are grouped in [milestones](https://github.com/falcosecurity/falco/milestones), the milestone with the next version represents what is going to be released.
+## Overview
+
+This document provides the process to create a new Falco release. In addition, it provides information about the versioning of the Falco components. At a high level each Falco release consists of the following main components:
+
+- Falco binary (userspace)
+- Falco kernel driver object files (kernel space)
+    - Option 1: Kernel module (`.ko` files)
+    - Option 2: eBPF (`.o` files)
+- Falco config and rules `.yaml` files (userspace)
+- Falco plugins (userspace - optional)
```

Comment:
Actually, some plugins (`k8saudit` and `cloudtrail`) are shipped together with Falco. 
That indirectly implies:
- plugins have their own versioning and [release strategy](https://github.com/falcosecurity/plugins/blob/master/release.md)
- If necessary, those plugins must be released before the Falco release, then their versions bumped in the related [cmake module](https://github.com/falcosecurity/falco/blob/master/cmake/modules/plugins.cmake) (it's similar to what we do with `libs` and `drivers`)
- Versioning: the plugin API version must be compatible